### ISSUE TITLE
Some certificates return an incomplete EKU list

### ISSUE THREAD

**Issue by christopher-henderson** [mozilla/tls-observatory#289]
tl;dr it may be worthwhile to rescan, at least, these certificates.

The certificates listed below are returning EKU lists with missing OIDs. Specifically:

```
1.3.6.1.4.1.311.10.3.12
1.3.6.1.4.1.311.10.3.4
1.3.6.1.4.1.311.21.5
2.16.840.1.113733.1.8.1
2.16.840.1.113741.1.2.3
```

There's nothing special about these OIDs that the Observatory can't figure out since I can [make and parse this certificate that appears to be handled just fine](https://tls-observatory.services.mozilla.com/static/certsplainer.html?id=186621556).  I'm currently thinking that these certs were scanned and cached when [this code](https://github.com/mozilla/tls-observatory/blame/8b54873f91d3705b1be2c9c1268865495d5cb56d/certificate/certificate.go#L284) was in place, which ignored the `UnknownExtKeyUsage`. If we rescan, at least these certs, I reckon that the EKUs will show up.

0CEBF97D1FABC64753799F7A9A508C7C5F2B58B928FB1B3CDC6C4109C0CF2E99
&nbsp;&nbsp;&nbsp;&nbsp;[crt.sh](https://crt.sh/?q=0CEBF97D1FABC64753799F7A9A508C7C5F2B58B928FB1B3CDC6C4109C0CF2E99)
&nbsp;&nbsp;&nbsp;&nbsp;[certsplainer](https://tls-observatory.services.mozilla.com/static/certsplainer.html?sha256=0CEBF97D1FABC64753799F7A9A508C7C5F2B58B928FB1B3CDC6C4109C0CF2E99)
1286173E6F0102F7BDD32C2F830910953489BF22C16295D84DD90A3DA137164A
&nbsp;&nbsp;&nbsp;&nbsp;[crt.sh](https://crt.sh/?q=1286173E6F0102F7BDD32C2F830910953489BF22C16295D84DD90A3DA137164A)
&nbsp;&nbsp;&nbsp;&nbsp;[certsplainer](https://tls-observatory.services.mozilla.com/static/certsplainer.html?sha256=1286173E6F0102F7BDD32C2F830910953489BF22C16295D84DD90A3DA137164A)
24AC889974694789DBD4E125FEE1A4BBA91532056795C05244BB66400A9D0316
&nbsp;&nbsp;&nbsp;&nbsp;[crt.sh](https://crt.sh/?q=24AC889974694789DBD4E125FEE1A4BBA91532056795C05244BB66400A9D0316)
&nbsp;&nbsp;&nbsp;&nbsp;[certsplainer](https://tls-observatory.services.mozilla.com/static/certsplainer.html?sha256=24AC889974694789DBD4E125FEE1A4BBA91532056795C05244BB66400A9D0316)
3625D9C482CD39E0DFBE8B76D7544F9FC93B78234B5F86C6CEDE2835EAA596A0
&nbsp;&nbsp;&nbsp;&nbsp;[crt.sh](https://crt.sh/?q=3625D9C482CD39E0DFBE8B76D7544F9FC93B78234B5F86C6CEDE2835EAA596A0)
&nbsp;&nbsp;&nbsp;&nbsp;[certsplainer](https://tls-observatory.services.mozilla.com/static/certsplainer.html?sha256=3625D9C482CD39E0DFBE8B76D7544F9FC93B78234B5F86C6CEDE2835EAA596A0)
3B9668F59F55FA3838FC2A3B80B7F9B5B13D1A46F1EAA6E0BCFF04C54198056C
&nbsp;&nbsp;&nbsp;&nbsp;[crt.sh](https://crt.sh/?q=3B9668F59F55FA3838FC2A3B80B7F9B5B13D1A46F1EAA6E0BCFF04C54198056C)
&nbsp;&nbsp;&nbsp;&nbsp;[certsplainer](https://tls-observatory.services.mozilla.com/static/certsplainer.html?sha256=3B9668F59F55FA3838FC2A3B80B7F9B5B13D1A46F1EAA6E0BCFF04C54198056C)
5587A72A6F738EDECEF67387116DE8D19370EAF3C6B272C49AF9BABA1795FD87
&nbsp;&nbsp;&nbsp;&nbsp;[crt.sh](https://crt.sh/?q=5587A72A6F738EDECEF67387116DE8D19370EAF3C6B272C49AF9BABA1795FD87)
&nbsp;&nbsp;&nbsp;&nbsp;[certsplainer](https://tls-observatory.services.mozilla.com/static/certsplainer.html?sha256=5587A72A6F738EDECEF67387116DE8D19370EAF3C6B272C49AF9BABA1795FD87)
6E21876CF16347201F637A178CB42B171D52379AF7E5F5B6F6755B3EE9BB2ED4
&nbsp;&nbsp;&nbsp;&nbsp;[crt.sh](https://crt.sh/?q=6E21876CF16347201F637A178CB42B171D52379AF7E5F5B6F6755B3EE9BB2ED4)
&nbsp;&nbsp;&nbsp;&nbsp;[certsplainer](https://tls-observatory.services.mozilla.com/static/certsplainer.html?sha256=6E21876CF16347201F637A178CB42B171D52379AF7E5F5B6F6755B3EE9BB2ED4)
6EDAD2261AA3D53BE0D6FF5FA71C34A34B253B0C637B35BF3E966540F78B631B
&nbsp;&nbsp;&nbsp;&nbsp;[crt.sh](https://crt.sh/?q=6EDAD2261AA3D53BE0D6FF5FA71C34A34B253B0C637B35BF3E966540F78B631B)
&nbsp;&nbsp;&nbsp;&nbsp;[certsplainer](https://tls-observatory.services.mozilla.com/static/certsplainer.html?sha256=6EDAD2261AA3D53BE0D6FF5FA71C34A34B253B0C637B35BF3E966540F78B631B)
719306D8FE2D9AB7BD17D5E211E439E50CE35B471F038B7E89BA098FFA5F2427
&nbsp;&nbsp;&nbsp;&nbsp;[crt.sh](https://crt.sh/?q=719306D8FE2D9AB7BD17D5E211E439E50CE35B471F038B7E89BA098FFA5F2427)
&nbsp;&nbsp;&nbsp;&nbsp;[certsplainer](https://tls-observatory.services.mozilla.com/static/certsplainer.html?sha256=719306D8FE2D9AB7BD17D5E211E439E50CE35B471F038B7E89BA098FFA5F2427)
7B464DC384FDB1A525C2CC279ED0C7CFAD24BECF72C46A7D7093D157C217607E
&nbsp;&nbsp;&nbsp;&nbsp;[crt.sh](https://crt.sh/?q=7B464DC384FDB1A525C2CC279ED0C7CFAD24BECF72C46A7D7093D157C217607E)
&nbsp;&nbsp;&nbsp;&nbsp;[certsplainer](https://tls-observatory.services.mozilla.com/static/certsplainer.html?sha256=7B464DC384FDB1A525C2CC279ED0C7CFAD24BECF72C46A7D7093D157C217607E)
89A7BFAB6B842542DCD9B043AE068483961F8E8603840FFEF93DDED30BB3CD7C
&nbsp;&nbsp;&nbsp;&nbsp;[crt.sh](https://crt.sh/?q=89A7BFAB6B842542DCD9B043AE068483961F8E8603840FFEF93DDED30BB3CD7C)
&nbsp;&nbsp;&nbsp;&nbsp;[certsplainer](https://tls-observatory.services.mozilla.com/static/certsplainer.html?sha256=89A7BFAB6B842542DCD9B043AE068483961F8E8603840FFEF93DDED30BB3CD7C)
8D162B501FE8720043FF31EF0919E2E40E50C99B80A8DD489155713212A65068
&nbsp;&nbsp;&nbsp;&nbsp;[crt.sh](https://crt.sh/?q=8D162B501FE8720043FF31EF0919E2E40E50C99B80A8DD489155713212A65068)
&nbsp;&nbsp;&nbsp;&nbsp;[certsplainer](https://tls-observatory.services.mozilla.com/static/certsplainer.html?sha256=8D162B501FE8720043FF31EF0919E2E40E50C99B80A8DD489155713212A65068)
96EF33C24A8B1F16CF170F43221E17E62AFF690A8B014F24529BFEB38F40A0DA
&nbsp;&nbsp;&nbsp;&nbsp;[crt.sh](https://crt.sh/?q=96EF33C24A8B1F16CF170F43221E17E62AFF690A8B014F24529BFEB38F40A0DA)
&nbsp;&nbsp;&nbsp;&nbsp;[certsplainer](https://tls-observatory.services.mozilla.com/static/certsplainer.html?sha256=96EF33C24A8B1F16CF170F43221E17E62AFF690A8B014F24529BFEB38F40A0DA)
ADDCB95B146C2E7742F16E22854FAA059D5CB87D406EC85EF7486694A03D6C84
&nbsp;&nbsp;&nbsp;&nbsp;[crt.sh](https://crt.sh/?q=ADDCB95B146C2E7742F16E22854FAA059D5CB87D406EC85EF7486694A03D6C84)
&nbsp;&nbsp;&nbsp;&nbsp;[certsplainer](https://tls-observatory.services.mozilla.com/static/certsplainer.html?sha256=ADDCB95B146C2E7742F16E22854FAA059D5CB87D406EC85EF7486694A03D6C84)
B2745D0A9B71D948F34E921AF59F342ADF6E407D88BC51D38AC52B583A0EBD15
&nbsp;&nbsp;&nbsp;&nbsp;[crt.sh](https://crt.sh/?q=B2745D0A9B71D948F34E921AF59F342ADF6E407D88BC51D38AC52B583A0EBD15)
&nbsp;&nbsp;&nbsp;&nbsp;[certsplainer](https://tls-observatory.services.mozilla.com/static/certsplainer.html?sha256=B2745D0A9B71D948F34E921AF59F342ADF6E407D88BC51D38AC52B583A0EBD15)
CF73B52D041B7309B439D16247414B90C9D26E44E38748A36500D5829B5187F9
&nbsp;&nbsp;&nbsp;&nbsp;[crt.sh](https://crt.sh/?q=CF73B52D041B7309B439D16247414B90C9D26E44E38748A36500D5829B5187F9)
&nbsp;&nbsp;&nbsp;&nbsp;[certsplainer](https://tls-observatory.services.mozilla.com/static/certsplainer.html?sha256=CF73B52D041B7309B439D16247414B90C9D26E44E38748A36500D5829B5187F9)
F74DA4D9A687C5046499AEFC9FA867DE91A1A3941A09F24CF94603B7AB78108A
&nbsp;&nbsp;&nbsp;&nbsp;[crt.sh](https://crt.sh/?q=F74DA4D9A687C5046499AEFC9FA867DE91A1A3941A09F24CF94603B7AB78108A)
&nbsp;&nbsp;&nbsp;&nbsp;[certsplainer](https://tls-observatory.services.mozilla.com/static/certsplainer.html?sha256=F74DA4D9A687C5046499AEFC9FA867DE91A1A3941A09F24CF94603B7AB78108A)

**Comment by jcjones** [comment]
Julien: Can we get these certs in the certsplainer links purged and rescanned?

**Comment by jvehent** [comment]
I wrote a small tool to reprocess some certs and generate the diffs. Could you take a look before I run the update? You can ignore changes that are not in the extensions. https://gist.github.com/jvehent/69fa883c9f3fdcb1435388e1c910532e

**Comment by jvehent** [comment]
if you're curious about the tool: https://github.com/mozilla/tls-observatory/blob/master/tools/updateCertificate.go
